torch/csrc/jit/codegen/cuda/partition.cpp

```diff
@@ -71,6 +71,222 @@ bool hasReductionOperation(const Node* node) {
   return false;
 }
 
+// utility function to check if the node implies broadcast on a given shape (
+// assumed to be shape of an input tensor)
+// limitations:
+//   1. we rely on shape information to judge this. so we would require output
+//      shape to be available;
+//   2. we basically compares given shape to the shape of the only output of
+//      the node and return true if it implies broadcast from the former to the
+//      latter.
+bool maybeBroadcastOnShape(
+    const Node* n,
+    const std::vector<c10::optional<int64_t>>& shape) {
+  TORCH_INTERNAL_ASSERT(
+      n->outputs().size() == 1,
+      "not expecting multiple outputs from a node, graph partitioning logic needs to be updated");
+  // assumes that if output is not a tensor type, it's not broadcasting
+  if (auto out_type = n->output(0)->type()->cast<TensorType>()) {
```

Comment:
Is this line right?

Reply:
I don't spot anything wrong with it.
We assert on single output the line above and we check to see if the output is `TensorType` before we start checking broadcasting.

Are you suggesting to assert on `TensorType` instead? I didn't do that in the hope of making the code future proof-ish, in the case we might have nodes that extract information of the tensor like `aten::size()`